src/Repo_old.java

```diff
@@ -0,0 +1,804 @@
+        final String proj = '/' + url.getProjectName() + '/';
+        int ind = src.indexOf(proj);
+
+        if (ind != -1) {
+            String view = src.substring(ind + proj.length());
+
+            ind = view.indexOf('/');
+
+            if (ind != -1) {
+                view = view.substring(0, ind);
+            }
+
+            return view;
+        }
+
+        return "main";
+    }
+
+    private static Folder getRootFolder(final StarTeamURL url, final View v) {
+        Folder f = v.getRootFolder();
+        boolean skipViewName = true;
+
+        stLoop: for (final StringTokenizer st = url.getFolders(); st
+                .hasMoreTokens();) {
+            final String name = st.nextToken();
+
+            if (skipViewName) {
+                skipViewName = false;
+                if (name.equals(v.getName())) {
+                    continue;
+                }
+            }
+
+            for (final Folder sub : f.getSubFolders()) {
+                if (name.equals(sub.getName())) {
+                    f = sub;
+                    continue stLoop;
+                }
+            }
+
+            throw new IllegalArgumentException("Invalid url '"
+                    + url.getSource() + "': folder " + name
+                    + " does not exist.");
+        }
+
+        return f;
+    }
+
+    private int autoLogOn(final Server server) {
+        try {
+            // Without synchronized autoLogOn() sometimes fails on concurrent
+            // access.
+            synchronized (Repo.class) {
+                return server.autoLogOn();
+            }
+        } catch (final Throwable ex) {
+            if (_logger.isDebugEnabled()) {
+                _logger.error(ex.getMessage(), ex);
+            }
+            return 0;
+        }
+    }
+
+    private static void cacheLogOnCredentials(final Server server,
+            final String userName, final String password) {
+        try {
+            server.cacheLogOnCredentials(userName, password);
+        } catch (final Throwable ex) {
+        }
+    }
+
+    private StarTeamURL getUrl() {
+        final String url = getRepoProperties().getProperty(PROP_URL);
+
+        if (url.startsWith("st::")) {
+            return new StarTeamURL(url.substring(4));
+        } else {
+            return new StarTeamURL(url);
+        }
+    }
+
+    private static String getProtocol(final StarTeamURL url) {
+        switch (url.getProtocol()) {
+        case PROTOCOL_TCP_IP_SOCKETS_XML:
+            return "starteam:xml";
+        default:
+            return "starteam";
+        }
+    }
+
+    private final class ConnectionPool extends SimpleObjectPool<Server> {
+
+        @Override
+        protected Server create() {
+            return createNewConnection();
+        }
+
+        @Override
+        protected void destroy(final Server s) {
+            s.disconnect();
+        }
+    }
+
+    private static final class DeleteHook extends Thread {
+        private final File _file;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "_file" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00116)